pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java

```diff
@@ -241,7 +241,11 @@ public static String replaceSpecialCharacterInPropertyValue(String value) {
    * {@link #replaceSpecialCharacterInPropertyValue(String)}.
    */
   public static String recoverSpecialCharacterInPropertyValue(String value) {
-    value = StringEscapeUtils.unescapeJava(value);
+    try {
+      value = StringEscapeUtils.unescapeJava(value);
+    } catch (Exception e) {
+      // If the value is not a valid escaped string, ignore the exception and continue
```

Comment:
Can we log an error message/exception here ?

Reply:
I don't think we want to log error here. If we want to find all old segments that contains unescaped special character, a metric might be helpful.
IMO we can just leave it as is since the overhead is minimal.